GOVERNANCE.md: remove “Community Processes” in favor of “GOVERNANCE.md” (#21067)
RIOT-OS/RIOT: GOVERNANCE.md

# Governance of the RIOT Community

The RIOT community is dedicated to creating a free and open source operating system for the constrained Internet of Things [[RFC7228], [draft-ietf-iotops-7228bis]] based on open standards.

> This governance document explains how the project is run.

Reviewer: This document explains the governance of the project.

## Community Processes
The community around RIOT gathers many IoT developers and users from around the world, from the industry, from academia, and hobbyists.
The RIOT community is open to everyone.
Our channels to communicate can be found in our [contributing guidelines].
The community self-organizes using the roles described below.

## Roles
### Contributors
Contributors are people who contribute there work to RIOT.
This includes,

- of course, code contributions, but also
- bug reports,
- other issue reports,
- participation in technical as well as non-technical discussions, or
- organizational considerations.

Code contributions are very welcome, but in order to streamline and harmonize code quality, contributors must follow the [contributing guidelines].

### Maintainers

Among contributors, some have maintainer status, which consists in rights (write access to the [RIOT GitHub repository](https://github.com/RIOT-OS/RIOT/)) and duties.
The current maintainers can be found in the [maintainers list].

Maintainers are people who care about RIOT and want to help it grow and improve.

> A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate with the team, get the most knowledgeable people to review code and docs, contribute high-quality code, and follow through to fix issues (in code or tests).

Author: Just a reminder to myself, that this alternative does not get lost.

```suggestion
A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate with the team, get the most knowledgeable people to review code and docs, contribute high-quality code, or follow through to fix issues (in code or tests).
```

Maintainers may resign at any time if they feel that they will not be able to
continue fulfilling their project duties.

Maintainers may also be removed after being inactive, upon failure to fulfill their
Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.
Inactivity is defined as a period of very low or no activity in the project.
A yearly maintainer ping, an e-mail sent to inactive maintainers, determines if the maintainer is still willing to fulfill their project duties.
On failure to reply to the maintainer ping within the specified amount of time (usually a month), the maintainer will be removed.

### Release Managers

Release managers make sure the quarterly release comes out in time.
They are one or more maintainers which were appointed for a specific release by the Virtual Maintainer Assembly (VMA).
Their duties include setting the dates for feature freeze for the release, enforcing the feature freeze, coordinating the (mostly automated) tests of a release, writing the release notes and creating the tags defining the release and its release candidates.
The full set of tasks can be found in the document [Managing a Release].

> Their duties end once the release is out or if all bug-fixing point releases are out.

Author: Quoting from https://github.com/RIOT-OS/RIOT/pull/21067#discussion_r1878075450 as this is somewhat lost in truncation.

>> also, this is a bit unclear. How do we know that "all bug-fixing point releases are out"? Just before starting a new (soft/hard) feature freeze for the next release ?
>
> I think that is something we need to decide on. Because I don't know an answer to your questions.

Reviewer 2: "Their duties end once the next release is out." 
as this ends the bug-fix cycle

# Governance of the RIOT Community

The RIOT community is dedicated to creating a free and open source operating system for the constrained Internet of Things [[RFC7228], [draft-ietf-iotops-7228bis]] based on open standards.
This document explains the governance of the project.

<!-- TOC start -->
<!-- TOC start, TOC end comments are required to generate TOC in Doxygen properly -->
- [Values](#values)
- [Community Processes](#community-processes)
- [Roles](#roles)
  + [Contributors](#Contributors)
  + [Maintainers](#Maintainers)
  + [Release Managers](#Release-Managers)
  + [Admins](#Admins)
  + [GitHub Owners](#GitHub-Owners)
  + [Moderators](#Moderators)
- [Decision Making](#decision-making)
- [Meetings](#meetings)
- [Code of Conduct](#code-of-conduct)
- [Security Response Team](#security-response-team)
- [Modifying this Charter](#modifying-this-charter)
<!-- TOC end -->

## Values

The RIOT community embraces the following values:

* **Openness:** Communication and decision-making happens in the open and is discoverable for future reference.

>   As much as possible, all discussions and work events eventually take place in public forums and open repositories.

Reviewer 3: I understand this as "For your idea to be considered, an open-ended public discussion must take place. If you want to evolve and refine your idea in private first before going public, that is fine."

I can imagine that some people might not feel comfortable to share an idea without getting some support for the idea first.

>   As much as possible, all discussions and work events eventually take place in public forums and open repositories.

Author: Added some words on that in https://github.com/RIOT-OS/RIOT/pull/21067/commits/548c6dc67559e16ca61a9616da1ae9c44188fb78

* **Fairness:** All stakeholders have the opportunity to provide feedback and submit contributions, which will be considered on their merits.

* **Community over Product or Company:** Sustaining and growing our community takes
  priority over shipping code or sponsors' organizational goals.
  Each contributor participates in the project as an individual.

* **Inclusivity:** We innovate through different perspectives and skill sets, which can only be accomplished in a welcoming and respectful environment.

> * **Participation:** Responsibilities within the project are earned through participation, and there is a clear path up the contributor ladder into more responsible positions.

Author: I apply the patch for now. I think we can do the contribution ladder in a follow-up (see also https://github.com/RIOT-OS/RIOT/pull/21067#issuecomment-2542082020), but in case nobody picks this up, we have the text not referencing to something that's not there.

## Community Processes
The community around RIOT gathers many IoT developers and users from around the world, from the industry, from academia, and hobbyists.
The RIOT community is open to everyone.
Our channels to communicate can be found in our [contributing guidelines].
The community self-organizes using the roles described below.

## Roles
### Contributors
Contributors are people who contribute their work to RIOT.
This includes,

- of course, code contributions, but also
- bug reports,

> - other issue reports,

Author: See d02c8190f3. I think that has a better flow and also fixes your typo.

- participation in technical as well as non-technical discussions, or
- organizational considerations.

Code contributions are very welcome, but in order to streamline and harmonize code quality, contributors must follow the [contributing guidelines].

### Maintainers

Among contributors, some have maintainer status, which consists in rights (write access to the [RIOT GitHub repository](https://github.com/RIOT-OS/RIOT/)) and duties.
The current maintainers can be found in the [maintainers list].

The duties of a maintainer include the following: Maintainers are people who care about RIOT and want to help it grow and improve.

> A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate with the team, get the most knowledgeable people to review code and docs, contribute high-quality code, and follow through to fix issues (in code or tests).

Reviewer 4: Did we have this wording before? It basically makes "contribute high-quality code" a prerequisite and would exclude e.g., @jkarinkl or @waehlisch from being maintainers. Which is probably not the intention?

Author: No that seem to have come from the [GOVERNANCE-maintainer.md template](https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md#maintainers).

Author: Would it be more helpful to make the "and" an "or"? I actually like having some specifics on “what makes a good maintainer” in here.

Author: I'd be happy to add any proposal :-)

Author: ```suggestion
A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate and organize with the team, get the most knowledgeable people to review code or documentation, contribute high-quality code or documentation, as well as follow through to fix issues (in code, tests, or tests).
```

would be my edit to that.

A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate with the team, get the most knowledgeable people to review code and docs, contribute high-quality code, and follow through to fix issues (in code or tests).
More on maintaining RIOT can be found in the [maintaining guidelines].

We are constantly looking for more maintainers.
So if you are up for that, please start (or continue) contributing code and reviews!

To contact maintainers, the best is to interact over actual RIOT code on [GitHub](https://github.com/RIOT-OS/RIOT/pulls).

#### Becoming a Maintainer

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Author: In the past it was very obvious in their behavior if a contributor intended to be a maintainer (reviewing PRs even without the rights to merge and writing issues), but in general there is no precedence of a would-be maintainer outright asking. So I'd say let's keep it as is, but I will put this in my forum summary of the current state of discussion.

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Reviewer 3: Hmm, let's say a contributor would approach a maintainer to also become a maintainer. I would assume that this would then be raised to the other maintainers and discussed pretty much in the same way as if the a maintainer suggested it.

Or would anyone of us reject such a request right away without reaching out to the others?

If not, we could also IMO just document that contributors may reach out to a maintainer of their choice, who will forward this to the other maintainers for consideration.

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Author: > If not, we could also IMO just document that contributors may reach out to a maintainer of their choice, who will forward this to the other maintainers for consideration.

Question is: is this something we need to put in our governance document, i.e., do we want to make it “law” that this is a way to become a maintainer? This could go into the contribution ladder document @jkarinkl proposed and would probably be a better fit.

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Reviewer 5: I would recommend keeping the governance doc as short as possible, so indeed, add this  to the contribution ladder. This part is about being friendly and welcoming to enthusiastic contributors, making it easy to step up if they are capable and willing.

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Author: > This part is about being friendly and welcoming to enthusiastic contributors, making it easy to step up if they are capable and willing.

So to add or not to add a sentence?

> Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

Author: I even added two sentences now ;-) b0e8500692

> If there is consensus on granting the status to a particular contributor, a maintainer will contact personally this contributor to propose the status, which the contributor can then accept (or turn down).

Author: This is usually done at the discretion of the maintainer that proposes the contributor.

A regular review exists for the “maintainer ping” (see section “Removing a Maintainer”), which does not have a set date, but usually happens around September/the RIOT summit.

> If there is consensus on granting the status to a particular contributor, a maintainer will contact personally this contributor to propose the status, which the contributor can then accept (or turn down).

Author: Added some words on that in https://github.com/RIOT-OS/RIOT/pull/21067/commits/548c6dc67559e16ca61a9616da1ae9c44188fb78.

Maintainers who are selected will be

- invited to the [maintainers GitHub team] by one of the admins of the RIOT project, which grants them the necessary GitHub rights,
- invited to the maintainer forum group by the forum moderators, which will give them access to the private maintainer forum, and

> - invited to the  private RIOT-maintainer chat room by one of the moderators of that room, for more informal exchanges between maintainers.

Reviewer 6: ```suggestion
- invited to the private RIOT-maintainer chat room by one of the moderators of that room, for more informal exchanges between maintainers.
```

- invited to the RIOT-maintainer chat room by one of the moderators of that room, for more informal exchanges between maintainers.

#### Removing a Maintainer

Maintainers may resign at any time if they feel that they will not be able to
continue fulfilling their project duties.

Maintainers may also be removed after being inactive, upon failure to fulfill their

> Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.

Author: I don't expect this to become an issue. But maybe others do?

> Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.

Reviewer 2: It might be a little to obvious but: actively, persistently, intentionally trying to (or succesfully) harm -especially the security or safety of- RIoT-OS might be a good reason to loose maintainer rights (either as contributor or maintainer). 

Removals for "other reasons" need to follow the process of decision making.

> Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.

Author: > It might be a little to obvious but: actively, persistently, intentionally trying to (or succesfully) harm -especially the security or safety of- RIoT-OS might be a good reason to loose maintainer rights (either as contributor or maintainer).

This is already covered by our Code of Conduct. In this case the maintainer would not just loose their maintainer right, but would also be put on hold on the RIOT communication channels.

> maybe we should state: removals for "other reasons" need to follow the process of decision making. (as "other reasons" are an exceptionally low bar)

Maybe we should just remove the “for other reasons” wording altogether?

> Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.

Reviewer 2: > This is already covered by our Code of Conduct. In this case the maintainer would not just loose their maintainer right, but would also be put on hold on the RIOT communication channels.

Isn't the Code of Conduct about communication rather than the OS (code) itself, I thought this (my suggestion) is more about the OS and Code rather than the communication (more like (try) introducing backdoor/ harm the product rather than the community)

> Maintainer responsibilities, because of violating the Code of Conduct, or for other reasons.

Author: Ah. Yes, this makes sense.

A yearly maintainer ping, an e-mail sent to inactive maintainers, determines if the maintainer is still willing to fulfill their project duties.
On failure to reply to the maintainer ping within the specified amount of time (usually a month), the maintainer will be removed.

### Release Managers

Release managers make sure the quarterly release comes out in time.
They are one or more maintainers which were appointed for a specific release by the Virtual Maintainer Assembly (VMA).
Their duties include setting the dates for feature freeze for the release, enforcing the feature freeze, coordinating the (mostly automated) tests of a release, writing the release notes and creating the tags defining the release and its release candidates.
The full set of tasks can be found in the document [Managing a Release].

> Their duties end, once the release is out or if all bug-fixing point releases are out.

Reviewer 6: ```suggestion
Their duties end once the release is out or if all bug-fixing point releases are out.
```

also, this is a bit unclear. How do we know that "all bug-fixing point releases are out"? Just before starting a new (soft/hard) feature freeze for the next release ?

Author: > also, this is a bit unclear. How do we know that "all bug-fixing point releases are out"? Just before starting a new (soft/hard) feature freeze for the next release ?

I think that is something we need to decide on. Because I don't know an answer to your questions.

Author: Moved to https://github.com/RIOT-OS/RIOT/pull/21067#discussion_r1878098710 as this got somewhat lost in GitHubs truncation.

Their duties end once the release is out or if all bug-fixing point releases are out.

### Admins

GitHub admins are a special subgroup among RIOT maintainers.
They are marked as such in the [maintainers list].
They have more access rights to the RIOT repositories, such as granting access to a repository, adding new members to a team, or enabling protection for Git branches.
Release managers might need to contact GitHub admins to configure the branch protection rules for the release branch.
Beyond those technical duties and access rights, they do not have any special rights among maintainers.

> They are picked by the maintainers, usually based on seniority.

Author: > What does seniority mean here?

A maintainer that worked for a certain time with the team. As to the specifics of “certain time”: Long enough so the other maintainers/admins know they can be trusted with these access rights.

> The maintainers try to take care to spread the admin responsibility over several project stakeholders within the maintainer body.

Author: Something like 30a2f854c?

> The maintainers try to take care to spread the admin responsibility over several project stakeholders within the maintainer body.

Reviewer 5: nice!

> The maintainers try to take care to spread the admin responsibility over several organizations within the maintainer body.

Author: I used "organizations" throughout the text. Intuitively, I would not know what is meant with “entities” and “institutions”, in my opinion, would exclude, e.g., independent people who just work on third-party funding.

> The maintainers try to take care to spread the admin responsibility over several organizations within the maintainer body.

Reviewer 7: I see what you mean, but "organization" also exclude non-affiliated people? What's an organization of 1 ;)
The "balance of power" subtly hinted at here seems fuzzy. How about something like "spread admin responsibilities over the different stakeholders" ?

> The maintainers try to take care to spread the admin responsibility over several organizations within the maintainer body.

Author: I am fine with stakeholders. See 15934ea36a.

PS: It's not just “balance of power” in the ephemeral sense, but also to reduce bus factors.

> There are also admins on the other RIOT discussion platforms, beyond technical administrative duties they do not have any special rights.

Reviewer 6: ```suggestion
There are also admins on the other RIOT discussion platforms. Beyond technical administrative duties they do not have any special rights.
```

There are also admins on the other RIOT discussion platforms. Beyond technical administrative duties they do not have any special rights.
These admins usually are appointed or self-appointed on merit, i.e., whoever sets up the platform usually is (one of) its admin(s).

### GitHub Owners
Github owners are a special subgroup among RIOT GitHub admins.
They are marked as such in the [maintainers list].
Beyond this special status and the usual GitHub admin rules and duties they do not have any special rights among maintainers.

### Moderators

Moderators are responsible to enforce the values of the RIOT community within its discussion platforms. Each platform usually has its own set of moderators, a list of which can be found there.

> The forum moderators, e.g., can be found [here](https://forum.riot-os.org/g/moderators) (link requires you to be logged into the forum).

Author: Also lost in trunctation https://github.com/RIOT-OS/RIOT/pull/21067#discussion_r1878076636:

>> We currently don't have any appointed moderators for other communication channels, do we? This text makes it sound as if we had (or at least would like to have) them, but doesn't give a hint on what those are.
> No, we don't have.
> 
Also somewhat lost in 

> With the chats it usually goes the “IRC way”. People are appointed somewhat on merrit and on how active they are. Since the moderation roles in the chats is, however, a little bit different than on the forum, I don't think there needs to be a moderator appointment procedure for that.
> 
> With the few mailing lists remaining, the moderators are mostly sorting out spam, so while similar to forum moderation, I also don't think appointment procedures are necessary here.
> 
> A forum moderator, on the other hand, actually can moderate a discussion on the forum, in the traditional sense of the word. They can edit posts, they can split out conversations in new topics and make forum topics a “wiki post” (so something everyone can edit). In that regard, they have much more power and require a certain level of trust from the community.

> The forum moderators, e.g., can be found [here](https://forum.riot-os.org/g/moderators) (link requires you to be logged into the forum).

Reviewer 6: We currently don't have any appointed moderators for other communication channels, do we? This text makes it sound as if we had (or at least would like to have) them, but doesn't give a hint on what those are.

Author: We do. The forum moderators where appointed by the community back when the forum was started.

>   decisions (a king or president), nor should decisions be made by a
>   vote, nor do we want decisions to be made in a vacuum without
>   practical experience.  Instead, we strive to make our decisions by
>   the consent of all participants, though allowing for some dissent
>   (rough consensus), and to have the actual products of engineering
>   (running code) trump theoretical designs.
>
>   Having full consensus, or unanimity, would be ideal, but we don't
>   require it: Requiring full consensus allows a single intransigent
>   person who simply keeps saying "No!" to stop the process cold.  We
>   only require rough consensus: If the chair of a working group
>   determines that a technical issue brought forward by an objector has
>   been truly considered by the working group, and the working group has
>   made an informed decision that the objection has been answered or is
>   not enough of a technical problem to prevent moving forward, the
>   chair can declare that there is rough consensus to go forward, the
>   objection notwithstanding.

Within the RIOT community, the duties of an IETF working group chair fall to maintainers knowledgeable in the area of expertise.

> This knowledgability is determined by their own contributions.

Author: > How exactly is decided who this is? As in: if there is disagreement on a topic, how is decided who the maintainers knowledgeable the area of expertise is, exactly? Can that be described here?

Within the IETF there working groups with one chair are uncommon. So why can't multiple maintainers be the deciders here as well? Rough consensus means that the opinions of experts also should not be disregarded, so this describes exactly this case.

> If there is disagreement on who this person is, what can be used as tiebreaker to decide?
> (to answer this question it helps to ask yourselves: what did successfully work in the past?)

So far a tiebreaker was not needed, if I remember correctly.

On decisions regarding a release, the release manager(s) take this position.

> 

Author: > How does this work with decision regarding documentation?

Documentation is typically about something code related, so I'd say the maintainer responsible for the code (after all, they could e.g. also see immediately that the doc is factually incorrect).

> And regarding strategy setting/vision/goals?

We have our maintainers / experts there as well (see https://github.com/RIOT-OS/RIOT/pull/21067#discussion_r1878064134 ff). :-)

> ## Meetings

Author: GA is not just for maintainers. The current wording of this section reflects this change. There are also discussions to open the VMA to non-maintaining participants. But as a reminder: This document aims to document the current status quo, improvements to the status quo should come as a follow-up.

> There are 3 types of meetings that maintainers may participate in:

Reviewer 6: ```suggestion
There are three types of regular meetings within the RIOT community:
```

(as GA is not restricted to maintainers)

> - The annual General Assembly (GA),
> - The quarterly Virtual Maintainer Assembly (VMA), and
> - and the Weekly Coordinational Meeting.

Reviewer 6: ```suggestion
- The annual General Assembly (GA),
- The quarterly Virtual Maintainer Assembly (VMA), and
- The Weekly Coordinational Meeting.
```

The GA is public and anyone who sees themselves as a member of the RIOT community can participate.
Larger community steering decisions for the community are made during the GA, e.g., electing contact people for the Code of Conduct.
The GA usually takes place during the [RIOT Summit](https://summit.riot-os.org/), the annual get-together of the RIOT community.
The GA moderator usually is appointed by the organizers of the RIOT Summit.
It is usually recorded and its notes will be published publicly in the RIOT forum.
The agenda for the GA is collected before the assembly but may be bashed at the start of the meeting.

The Virtual Maintainer Assembly (VMA) is a closed meeting among maintainers.
The VMA appoints the release manager for upcoming releases and the moderator for the next VMA.
Other maintenance decisions such as the fate of larger sections of code are discussed after these administrative tasks are done.

> The VMA usually takes place about a month after the latest release, usually in a virtual space, such as a video conference, the VMA moderator polls the maintainers for a sufficient date around the date of the upcoming release.

Reviewer 6: ```suggestion
The VMA usually takes place about a month after the latest release, usually in a virtual space, such as a video conference.
The VMA moderator polls the maintainers for a sufficient date around the date of the upcoming release.
```

> The VMA may or may not co-incide with the GA, usually the VMA moderator decides, however, to merge these two meetings.

Reviewer 6: ```suggestion
Every forth VMA may or may not co-incide with the GA. However, the VMA moderator usually decides to merge these two meetings.
```

In this case, VMA moderator and RIOT Summit organizers decide together on who is moderating the joint event (it may be the VMA moderator, it may be someone else).
The agenda for the VMA is collected before the assembly but may be bashed at the start of the meeting.

> The notes of the VMA will be published publicly in the RIOT forum.

Reviewer 6: ```suggestion
The notes of the VMA will be published publicly in the [RIOT forum](https://forum.riot-os.org/tag/vma).
```

Author: I intentionally try to avoid links to platforms beyond our website or GitHub in this document, as those might change in the future, while putting them in here would just ossify them (or the links could spoil like milk,... see the old maintainer list or other links [I just had to move around](https://github.com/RIOT-OS/RIOT/pull/21066)), see also https://github.com/RIOT-OS/RIOT/pull/21067#discussion_r1878063875.

Reviewer 2: to avoid spoiling we could link a reference document and links to people lists (maintainer, moderator-list .. and covers link like this)

Author: We have already the README and many other documents (CONTRIBUTING.md to name one) that have such lists. Do we really need yet another one?

Reviewer 2: right, i thought about the other link in this document but avoiding more links is probably better.

> The agenda for the VMA is collected before the assembly but may be bashed at the start of the meeting.

Reviewer 6: ```suggestion
```

sentence is doubled

The Weekly Coordinational Meeting is a closed meeting among maintainers.
It usually serves as a small communal get-together of maintainers on a regular basis.
Smaller maintenance decisions are made during these meetings, but also short term admistrative tasks are discussed.
The Weekly Coordinational Meeting usually takes place every Friday at 10:00 in a virtual space, such as a video conference.
A maintainer that feels responsible for it shares the link to the meeting as well as a proposed agenda, which may be amended by other maintainers, usually a day in advance.

## Code of Conduct

[Code of Conduct] violations by community members will be discussed and resolved on the conduct@riot-os.org list.

> If one of the appointees to that list (see GA notes for the members of that list) is involved in a Code of Conduct violation, two forum moderators from other organizations than the appointee take their place in the discussions.

Reviewer 6: > see GA notes for the members of that list

Could we have a clearer pointer and/or directly a list of appointees here?

Author: Having an actual list of people in this document should be avoided. Since changing this document is supposed to be hard, it would add a lot of administrative overhead to update that.

Reviewer 6: I see, so having _another_ place with a permanent link to check for the current appointees would be great.

Author: Maybe we should add them to https://www.riot-os.org/maintainers.html? @OlegHahm what do you think?

Author: Come to think of it... I am not sure this is a good idea. It gives the impression that only maintainers can be Code of Conduct contacts or that that role is somehow connected to be a maintainer in other ways. Since the list is not expected to be changing that much. Maybe we just should add them to [CODE_OF_CONDUCT.md](https://github.com/RIOT-OS/RIOT/blob/master/CODE_OF_CONDUCT.md)?

Author: See https://github.com/RIOT-OS/RIOT/wiki/CoC:-Reporting-Guidelines and https://github.com/RIOT-OS/RIOT/pull/21071 (waiting with updating this PR until either #21071 or we decide to stick to the wiki)

## Security Response Team

> The Maintainers will appoint a Security Response Team to handle security reports.

Reviewer 6: ```suggestion
The maintainers will appoint a Security Response Team to handle security reports.
```

It seems to be lowercase everywhere else, too.

This committee may simply consist of the maintainers themselves.

> 

Reviewer 6: ```suggestion
```

The Security Response Team is responsible for handling all reports of security holes and breaches according to the [security policy].

## Modifying this Charter

Changes to this Governance and its supporting documents require the approval of at least four maintainers who all must be employed by or associated with different stakeholders.

> 

Author: See 8204ce185c

> Changes to this Governance and its supporting documents require the approval of at least 4 maintainers who all must be employed by or associated with different organizations.

Reviewer 6: ```suggestion
Changes to this Governance and its supporting documents require the approval of at least four maintainers who all must be employed by or associated with different organizations.
```

The community self-organizes using the roles described below.

## Roles
### Contributors
Contributors are people who contribute their work to RIOT.
This includes,

- of course, code contributions, but also
- writing documentation,
- bug reports,
- other issue reports,
- reviewing PRs,
- participation in technical as well as non-technical discussions, or
- organizational considerations.

Code contributions are very welcome.
In order to streamline and harmonize code quality, contributors must follow the [contributing guidelines].

Contributors may be associated with organizations—by employment or otherwise—who have a vested interest in RIOT or may be individuals who have their own personal stakes in RIOT.
We call these organizations and individuals “stakeholders” throughout this document to summarize them.

### Maintainers

Among contributors, some have maintainer status, which consists in rights (write access to the [RIOT GitHub repository](https://github.com/RIOT-OS/RIOT/)) and duties.
The current maintainers can be found in the [maintainers list].

Maintainers are people who care about RIOT and want to help it grow and improve.

> A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate and organize with the team, get the most knowledgeable people to review code or documentation, contribute high-quality code or documentation, as well as follow through to fix issues (in code, tests, or tests).

Reviewer 2: other .md files [CoC](https://raw.githubusercontent.com/RIOT-OS/RIOT/refs/heads/master/CODE_OF_CONDUCT.md) seem to have some line length limit applied  (maybe 100 with url exception)

But you are probably just saving this for the squash

Author: TBH, from other documents, I prefer the “one sentence, one line” style, since it makes it easier to track and merge changes (editors can be configured to auto wrap text which is not as ugly as e.g. with code), but I'm fine with fixing it to 100 characters max.

### Maintainers

Among contributors, some have maintainer status, which consists in rights (write access to the [RIOT GitHub repository](https://github.com/RIOT-OS/RIOT/)) and duties.
The current maintainers can be found in the [maintainers list].

Maintainers are people who care about RIOT and want to help it grow and improve.
A maintainer is not just someone who can make changes, but someone who has demonstrated their ability to collaborate and organize with the team, get the most knowledgeable people to review code or documentation, contribute high-quality code or documentation, as well as follow through to fix issues (in code, tests, or tests).
More on maintaining RIOT can be found in the [maintaining guidelines].

We are constantly looking for more maintainers.
So if you are up for that, please start (or continue) contributing code and reviews!

To contact maintainers, the best is to interact over actual RIOT code on [GitHub](https://github.com/RIOT-OS/RIOT/pulls).

#### Becoming a Maintainer

Maintainers can propose to give maintainer status to contributors that have been noticed as particularly active in some domain of RIOT.

> Of course, contributors also can step forward themselves and declare their interest to become a maintainer.
> In this case, maintainers should then propose them as well, if there activities can be confirmed.

Reviewer 6: ```suggestion
Of course, contributors can also step forward themselves and declare their interest to become a maintainer.
In this case, maintainers should then propose them as well, if their activities can be confirmed.
```
